pkg/client/actionconfig.go

```diff
@@ -30,72 +32,76 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	v1 "k8s.io/client-go/kubernetes/typed/core/v1"
 	"k8s.io/client-go/rest"
-	cmdutil "k8s.io/kubectl/pkg/cmd/util"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
 type ActionConfigGetter interface {
 	ActionConfigFor(obj client.Object) (*action.Configuration, error)
 }
 
-func NewActionConfigGetter(cfg *rest.Config, rm meta.RESTMapper, log logr.Logger) ActionConfigGetter {
-	return &actionConfigGetter{
-		cfg:        cfg,
-		restMapper: rm,
-		log:        log,
-	}
+type errActionConfigGetter struct {
```

Comment:
Can you add a comment why an `errActionConfigGetter` is introduced?
It may not be obvious to a reader to understand the reasoning that this prevents an OOM crash in cases where lots of reconciles happens like errors.

Reply:
Well, actually I removed the `errActionConfigGetter`, and the reasoning to introduce it was not to prevent OOMs (the other fixes accomplish that).

The reason I introduced an `errActionConfigGetter` was that it seemed inconvenient to change the signature of this function to return an `error` in addition to the `ActionConfigGetter`, especially due to it being used in tests. However, after I gave it a second look it turned out to be not that big of a deal, so I changed the signature to return an `error` and no longer need an `errActionConfigGetter`.